posts/2023-11-14-23.0.0.11.adoc

```diff
@@ -0,0 +1,243 @@
+vendor_connectionpool_inUseTime_per_usedConnection_seconds{datasource="jdbc_exampleDS1"} 0.743`
+
+|Connection Pool Wait Time per Queued Request
+|`# HELP connectionpool_waitTime_per_queuedRequest_seconds The recent average wait time for queued connection requests. +
+# TYPE connectionpool_waitTime_per_queuedRequest_seconds gauge connectionpool_waitTime_per_queuedRequest_seconds{datasource="jdbc_exampleDS1",mp_scope="vendor",} 35.0`
+|`# TYPE vendor_connectionpool_waitTime_per_queuedRequest_seconds gauge +
+# HELP vendor_connectionpool_waitTime_per_queuedRequest_seconds The recent average wait time for queued connection requests.
+vendor_connectionpool_waitTime_per_queuedRequest_seconds{datasource="jdbc_exampleDS1"} 45.0`
+
+|Servlet Elapsed Time per Request
+|`# HELP servlet_request_elapsedTime_per_request_seconds The recent average elapsed response time per servlet request. +
+# TYPE servlet_request_elapsedTime_per_request_seconds gauge servlet_request_elapsedTime_per_request_seconds{mp_scope="vendor",servlet=”myapp_servletA",} 0.001256676333333333
+servlet_request_elapsedTime_per_request_seconds{mp_scope="vendor",servlet=" myapp_servletB",} 0.00372855566666666
+servlet_request_elapsedTime_per_request_seconds{mp_scope="vendor",servlet=" myapp_servletC",} 1.731813674`
+|`# TYPE vendor_servlet_request_elapsedTime_per_request_seconds gauge +
+# HELP vendor_servlet_request_elapsedTime_per_request_seconds The recent average elapsed response time per servlet request. 
+vendor_servlet_request_elapsedTime_per_request_seconds{servlet=”myapp_servletA",} 0.36816000695238094
+vendor_servlet_request_elapsedTime_per_request_seconds{servlet=" myapp_servletB",} 0.384967335
+vendor_servlet_request_elapsedTime_per_request_seconds{servlet=" myapp_servletC",} 1.333671328`
+
+|REST Elapsed Time per Request
+|`# HELP REST_request_elapsedTime_per_request_seconds The recent average elapsed response time per RESTful resource method request. +
+# TYPE REST_request_elapsedTime_per_request_seconds gauge REST_request_elapsedTime_per_request_seconds{class=”my.package.MyClass",method=”simpleGet",mp_scope=”vendor"} 0.0061460695`
+|`# TYPE vendor_REST_request_elapsedTime_per_request_seconds gauge +
+# HELP vendor_REST_request_elapsedTime_per_request_seconds The recent average elapsed response time per RESTful resource method request. 
+vendor_REST_request_elapsedTime_per_request_seconds{class=”my.package.MyClass",method=”simpleGet"} 0.0024352581`
+
+|===
+
+
+The Heap Utilization and CPU Utilization metrics are available when the server is started. The Connection Pool, REST, and Servlet metrics are available if the application contains any of the relevant data sources, REST APIs, or servlets, as is the case with the existing vendor metrics.
+
+The new vendor metrics are available in the `/metrics` output, by enabling the following Microprofile Metrics 5.0 feature in your `server.xml`:
```

Comment:
```suggestion
The new vendor metrics are available in the `/metrics` output when you enable the following Microprofile Metrics 5.0 feature in your `server.xml`:
```

Reply:
@pgunapal - do you have to enable 5.0? The heading says 3,4, and 5 and I beleive the new metrics are avaialble going back to 3.0. But not sure if you have to enable 5 or just "3 or later"

Reply:
Good catch. This is the text copied from the beta. I think it should be 3 or later.